examples/references/segmentation/pascal_voc2012/code/scripts/training.py

```diff
@@ -232,21 +237,33 @@ def run_validation():
                 trainer, optimizer, evaluators={"training": train_evaluator, "validation": evaluator}
             )
 
-        # Log val predictions:
+        # Log validation predictions as images
+        # We define a custom event filter to log less frequently the images (to reduce storage size)
+        # - we plot images with masks of the middle validation batch
+        # - once every 3 validations and
+        # - at the end of the training
+        def custom_event_filter(_, val_iteration):
```

Comment:
Remark : I will think about how implementing in a different way EventList to overcome this.